client/core/core.go

```diff
@@ -5241,8 +5244,10 @@ func (c *Core) Logout() error {
 	// Lock wallets
 	if !c.cfg.NoAutoWalletLock {
 		for _, w := range c.xcWallets() {
-			if w.connected() {
-				if err := w.Lock(5 * time.Second); err != nil {
+			if w.connected() && w.unlocked() {
+				symb := strings.ToUpper(unbip(w.AssetID))
+				c.log.Infof("Locking %s wallet", symb)
+				if err := w.Lock(walletLockTimeout); err != nil {
```

Comment:
Still looks like locking could race with shutdown. Could it just hold the `c.walletMtx` read lock as opposed to doing `c.xcWallets()`?

Reply:
@buck54321 suggested using `c.wg`, implemented in https://github.com/decred/dcrdex/pull/2781#issuecomment-2127750949

Reply:
Is it possible for this to happen after the `c.wg.Wait()` in run? It's probably unlikely and doesn't really matter in the end, so actually don't worry about it.

Reply:
> Is it possible for this to happen after the c.wg.Wait() in run

It happens after `c.wg.Wait` in `Run`.